src/main/java/nl/tudelft/dnainator/ui/DNAinator.java

```diff
@@ -24,20 +32,55 @@
 	private static final int MIN_WIDTH = 300;
 	private static final int MIN_HEIGHT = 150;
 
+	private Stage stage;
+	private BooleanProperty ready = new SimpleBooleanProperty(false);
+
+	private void loadDB() {
+		Task<Graph> task = new Task<Graph>() {
+			@Override
+			protected Graph call() throws Exception {
+				Graph db = Neo4jSingleton.getInstance().getDatabase();
+				ready.set(true);
+				return db;
+			}
+		};
+
+		new Thread(task).start();
+	}
+
+	@Override
+	public void init() {
+		loadDB();
+	}
+
 	@Override
 	public void start(Stage primaryStage) throws Exception {
-		primaryStage.setMinHeight(MIN_HEIGHT);
-		primaryStage.setMinWidth(MIN_WIDTH);
-		primaryStage.setTitle(DNAINATOR);
-		primaryStage.getIcons().add(new Image(getClass().getResourceAsStream(ICON)));
-		primaryStage.setMaximized(true);
-		try {
-			BorderPane rootLayout = FXMLLoader.load(getClass().getResource(FXML));
-			Scene scene = new Scene(rootLayout);
-			primaryStage.setScene(scene);
-		} catch (IOException e) {
-			e.printStackTrace();
+		ready.addListener(event -> {
+			Platform.runLater(() -> {
+				stage = primaryStage;
+				stage.setTitle(DNAINATOR);
+				stage.getIcons().add(new Image(getClass().getResourceAsStream(ICON)));
+				stage.setMaximized(true);
+
+				notifyPreloader(new Preloader.ProgressNotification(1));
+
+				try {
+					BorderPane rootLayout = FXMLLoader.load(getClass().getResource(FXML));
+					stage.setScene(new Scene(rootLayout));
+					stage.show();
+				} catch (IOException e) {
+					e.printStackTrace();
+				}
+			});
+		});
+	}
+
+	@Override
+	public void stop() throws Exception {
+		Set<String> paths = Neo4jSingleton.getInstance().getDatabasePaths();
```

Comment:
Stopping databases is not actually needed.
A shutdown listener is created for each db when it's started, which will cleanly shut it down anyways.
All this does right now is delete all databases when exiting, which might not be what we want.